src/ott/neural/data/semidiscrete_dataloader.py

```diff
@@ -0,0 +1,145 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+from typing import Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+import jax.random as jr
+
+from ott.solvers.linear import semidiscrete
+
+__all__ = ["SemidiscreteDataloader"]
+
+
+@dataclasses.dataclass(frozen=False, repr=False)
+class SemidiscreteDataloader:
+  """Semidiscrete dataloader.
+
+  The dataloader samples from the continuous source distribution and
+  couples it with the discrete target distribution. It returns aligned tuples of
+  ``(source, target)`` arrays of shape ``[batch_size, ...]``.
+
+  Args:
+    out: Semidiscrete output.
+    batch_size: Batch size.
+    rng: Random number seed.
+    subset_threshold: Threshold above which to sample from a subset of the
```

Comment:
`subset_size_threshold` ? usually `threshold` is used in ott-jax for a float, here it's an int